src/Control/Abstract/Value.hs

```diff
@@ -292,15 +290,25 @@ liftNumeric2 :: (Member (Numeric value) sig, Carrier sig m)
              -> value
              -> value
              -> m value
-liftNumeric2 t v1 v2 = send (LiftNumeric2 t v1 v2 pure)
+liftNumeric2 t v1 v2 = send (LiftNumeric2 (Numeric2Function t) v1 v2 pure)
+
+data NumericFunction = NumericFunction (forall a . Num a => a -> a)
```

Comment:
We had to introduce these to retain the derived implementations of the `Numeric` effect. I don’t object to it, though it means slightly more code.